app/validators/proxy_features_validator.rb

```diff
@@ -0,0 +1,15 @@
+class ProxyFeaturesValidator < ActiveModel::EachValidator
+  def initialize(args)
+    @features = args
+    @features.delete(:class)
+    super
+  end
+
+  def validate_each(record, attribute, value)
+    if value && !value.has_feature?(@features[attribute])
+      translated = _(@features[attribute])
+      msg = _("Smart Proxy %{a} does not have %{b} feature") % { :a => value.name, :b => translated }
```

Comment:
1. When using named string interpolation, use descriptive names (not 'a', 'b') so translators know what they represent.
2. Usually attribute errors in AR are written as partial sentences, e.g. "is too long", "can't be blank".  So consider changing this to just `is missing the DNS feature`, because it'll be recorded against a smart proxy dropdown field, so there's no need to repeat that information or its name.